.github/workflows/linux-qe-template.yml

```diff
@@ -0,0 +1,200 @@
+name: linux-qe-template
+
+on:
+  workflow_call:  
+    inputs:  
+      trigger-workflow-run-id:  
+        required: true  
+        type: string
+      qe-type:  
+        description: type of test; allowed values e2e or integration
+        required: true  
+        type: string
+      preset:
+        description: preset type only required if qe-type is e2e
+        type: string  
+
+jobs:
+  linux-qe:
+    runs-on: [self-hosted, linux, testing-farm]
+    permissions:
+      statuses: write # needed to update commit status (pending/failure/sucess)
+      checks: write # as documented in https://github.com/mikepenz/action-junit-report?tab=readme-ov-file#pr-run-permissions
+    steps:
+      - name: prepare env
+        run: |
+          sudo yum install podman openssh-server -y
+
+          commit_sha=$(cat gh_context.json | jq -r '.event.after')
+          if [[ -z "${commit_sha}" ]] || [[ "${commit_sha}" == null ]]; then
+            # on first PR creation .event.after is empty, then .sha is used as commit instead
+            commit_sha=$(cat gh_context.json | jq -r '.event.pull_request.head.sha')
+          fi
+          echo "commit_sha=${commit_sha}" >> "$GITHUB_ENV"
+          mkdir ${{ env.commit_sha }}-${{inputs.qe-type}}-${{inputs.preset}}
+          cd ${{ env.commit_sha }}-${{inputs.qe-type}}-${{inputs.preset}}
+
```

Comment:
May consider add some comments in here to explain why we need the socket strategy on the target..i.e. the target can only be accessed through a bastion (which can only be accessed from self-hosted runner) as so we need to map the ssh-agent from the host to the containers used to access the target host